Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -311,6 +326,20 @@ namespace O3DE::ProjectManager
         return gemSystemNameIsValid;
     }
 
+    bool CreateGem::ValidateGemPath()
+    {
+        if (m_gemLocation->lineEdit()->text().isEmpty())
+        {
+            return false;
+        }
+
+        QDir chosenGemLocation = QDir(m_gemLocation->lineEdit()->text());
+
+        bool locationValid = !chosenGemLocation.exists() || chosenGemLocation.isEmpty();
```

Comment:
how can you get the situation where a path makes it past the `isEmpty()` check on the `lineEdit` but could be empty after making it a `QDir`?  Does that happen if you enter in some non-path character as input like "?"

Reply:
The first isEmpty check is to check the string is empty - if it is then QDir is initialized as the current directory. It's avoids creating a gem in the current folder if you run the project manager from an empty folder and forget to fill out this field which would be unexpected from a GUI app. Very edge case however, so I wouldn't feel too strongly about removing the first check if it affect readability.

Reply:
thanks for clearing that up! might be worth a comment in the code, but unlikely many folks will go changing this

Reply:
A comment is always useful for unconventional , even if the code is unlikely the change.
It at least leaves a record in the future for another developer to understand the context of the code.

The only person who can accurately provide that context for future developers is the author and it is often appreciated by developers that they don't have to guess the author's intent.

Reply:
sometimes the author forgets why they changed something XD.